src/HVGroup.cpp

```diff
@@ -7,51 +10,94 @@ HVGroup::HVGroup(Interface& m_interface):
+            hventry.value_label->hide();
+            //connect(hventry.spin_box, static_cast<void (QSpinBox::*)(int)>(&QSpinBox::valueChanged), this, &HVGroup::valueChanged);
+            m_hventries.push_back(hventry);
+            m_layout->addWidget(hventry.label, temp_iterator, 0);
+            m_layout->addWidget(hventry.spin_box, temp_iterator, 1);
+            m_layout->addWidget(hventry.value_label, temp_iterator, 1);
+            temp_iterator++;
+        }
 
-        m_set = new QPushButton("Set");
+        m_on_btn = new QPushButton("Switch On");
+        //connect(m_on_btn, &QPushButton::clicked, [&](){ m_interface.getSetupManager()->switchHVPMTON(); });
+        connect(m_on_btn, &QPushButton::clicked, this, &HVGroup::switchON);
+        m_layout->addWidget(m_on_btn, temp_iterator, 0);
 
-        m_layout = new QGridLayout();
-        m_layout->addWidget(m_label, 0, 0);
-        m_layout->addWidget(m_spin, 0, 1);
-        m_layout->addWidget(m_value_label, 0, 1);
-        m_layout->addWidget(m_set, 0, 2);
-        m_box->setLayout(m_layout);
+        m_off_btn = new QPushButton("Switch OFF");
+        connect(m_off_btn, &QPushButton::clicked, this, &HVGroup::switchOFF);
+        m_layout->addWidget(m_off_btn, temp_iterator, 0);
+        m_off_btn->hide();
 
-        connect(m_spin, static_cast<void (QSpinBox::*)(int)>(&QSpinBox::valueChanged), this, &HVGroup::valueChanged);
+        m_set = new QPushButton("Set");
         connect(m_set, &QPushButton::clicked, this, &HVGroup::setHV);
+        m_layout->addWidget(m_set, temp_iterator, 1);
+        m_box->setLayout(m_layout);
+
 }
 
-void HVGroup::valueChanged(int m_hv) {
-    if(m_interface.isRunning()) {
-        m_spin->setValue(m_interface.getSetup().getHV());
-        return;
-    }
+//void HVGroup::valueChanged(int m_hv) {
+//    if(m_interface.isRunning()) {
+//        //m_spin_hv0->setValue(m_interface.getConditions().getHVPMTSetValues()[0]);
+//        return;
+//    }
+//}
+
+void HVGroup::switchON() {
+    m_interface.getSetupManager()->switchHVPMTON();
+    m_on_btn->hide();
+    m_off_btn->show();
 }
 
-void HVGroup::setHV() {
-    if(m_interface.isRunning())
-        return;
+void HVGroup::switchOFF() {
+    m_interface.getSetupManager()->switchHVPMTOFF();
+    m_off_btn->hide();
+    m_on_btn->show();
+}
 
-    m_interface.setHV(m_spin->value());
-    m_value_label->setText(QString::number(m_spin->value()));
+void HVGroup::setHV() {
+    // Update Condition manager with new HV set values
+    int temp_iterator = 0;
+    for (HVEntry hventry : m_hventries) {
+        std::ostringstream HVValue;
+        HVValue << hventry.spin_box->value();
+        hventry.value_label = new QLabel(HVValue.str().c_str());
+        m_interface.getConditions().setHVPMTValue(temp_iterator, hventry.spin_box->value());
+        temp_iterator++;
+    }
+    // Set physically the HV values
+    m_interface.getSetupManager()->setHVPMT();
```

Comment:
Same comments here about loop and strings...

Actually here there might be some thread concurrency issue, but this doesn't prevent to merge and have a deeper look afterwards.